### DISCUSSION TITLE
Neovim Copilot proxy is not working after 1.19.0 (a90fb6dd68a)

### DISCUSSION THREAD

**Discussion by npupko** [community/community#109683] [Copilot Conversations] [answered]
### Select Topic Area

Bug

### Body

After the release of [a90fb6dd68](https://github.com/github/copilot.vim/commit/a90fb6dd68a1c8fabd57e4df34dd471c27c18d66) my configuration just stopped working. Everything is fine if I will use the older release.

My configuration:
```lua
return {
  event = "VeryLazy",
  commit = "22fd9542e3c47552abab5e685fefd760e4f405b2",
  'github/copilot.vim',
  enabled = true,
  config = function()
    vim.g.copilot_no_tab_map = true
    vim.g.copilot_assume_mapped = true
    -- Digitalocean VPS
    vim.g.copilot_proxy = os.getenv("COPILOT_PROXY_URL")
    vim.keymap.set('i', '<M-\\>', '<Plug>(copilot-dismiss)', { silent = true })

    vim.g.copilot_filetypes = {
      markdown = true
    }
  end
}
```

```
:version
NVIM v0.9.5
Build type: Release
LuaJIT 2.1.1703358377
```

Where `COPILOT_PROXY_URL` is ENV in a format of `http://login:lassword@1.1.1.1:1111`

**Comment by tpope** [comment]
Hi! Can you check `:Copilot status` and `:Copilot version` and tell me if you see any errors?

Version 1.20.0 went out a couple of hours ago. Can you try that too?

**Reply by npupko** [reply]
@tpope output of the `:Copilot status` is `Copilot: Not authorized`
Versions are:
```
copilot.vim 1.20.1
Neovim 0.9.5
GitHub Copilot Language Server 1.165.0
Node.js 20.9.0
```

`:echo g:copilot_proxy` is in place and correct

It still is not working for me unfortunately. Last working version is v1.18.0

**Reply by tpope** [reply, marked as answer]
Do you have any `%` escapes in the username/password in your proxy URL?

**Reply by npupko** [reply]
@tpope Yep, I have an escaped `%` and `$` 🤔

**Reply by tpope** [reply]
That's probably it. Are they escaped correctly, as `%25` and `%24` respectively?

**Reply by tpope** [reply]
Also, if you check `:Copilot log`, you might find an error message about parsing a URI.

**Reply by npupko** [reply]
This fixed the problem, thank you very much. Both symbols were not escaped with `%`
